src/validation/coinbase.cairo

```diff
@@ -66,9 +67,47 @@ pub fn validate_coinbase(
         if witness != @"\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0" {
             return Result::Err("Wrong coinbase witness");
         }
-    }
 
-    // TODO: validate BIP-141 segwit output
+        // validate BIP-141 segwit output
+        if *tx.is_segwit {
+            let outputs = *tx.outputs;
+            let mut is_wtxid_commitment_present: bool = false;
+            let mut extracted_wtxid_commitment: ByteArray = "";
+            let mut i = 0;
+            let mut x = 6;
+
+            while i < outputs.len() {
+                let pk_script = *outputs[i].pk_script;
+
+                // check for OP_RETURN and the fixed prefix "0xaa21a9ed"
+                if pk_script.len() >= MINIMUM_WITNESS_COMMITMENT
```

Comment:
It is cheaper (and cleaner) to compare bytearrays rather than individual bytes (the more the length the more the difference).

So what you can do is to construct the array "6124aa21a9ed<commitment>" using concat, and then for every output create script substring of length 39 and compare with the expected one.